src/main/scala/DataMunging.scala

```diff
@@ -0,0 +1,40 @@
+package DataMunging
+import scala.io.Source
+
+object DataMunging {
+
+    def main(args: Array[String]) ={
+            getMinSpread()
+            getMinGoalDiff()
+    }
+    def getMinSpread() = {
+        val source = Source.fromFile("weather.dat")
+        val spread = source.getLines.toList
+                               .drop(2) //the first line is a header and second line is a blank space so skip those lines here.
+                               .map(_.split("\\s+")) // split on regex white space
+                               .map(x => (x(1), x(2).stripSuffix("*").toDouble-x(3).stripSuffix("*").toDouble)) //clean the recourds as somelines have * and aslo conver to double
+        val min_spread = find_min(spread) // get minimum spread
+        println(s"1: the day with the smallest temparature spread is day ${min_spread._1} with a spread of ${min_spread._2} degrees")
+        source.close()
+    }
+
+    def getMinGoalDiff() = {
+        val source = Source.fromFile("football.dat") //get file  access object
+        val teams = source.getLines.toList
+                         .drop(1) //drop the header
```

Comment:
If you implement getLines the way I suggest, getMinGoalDiff() becomes 
``` def getMinGoalDiff(line: Array[String):(String, Double)```. 
Also lines 22 to 26 will no longer be needed.